src/extensions/etags.ts

```diff
@@ -0,0 +1,42 @@
+import { RC } from "../rc";
+
+export function saveEtags(
+  rc: RC,
+  projectId: string,
+  instances: { instanceId: string; etag?: string }[]
+): void {
+  rc.setEtags(projectId, "extensionInstances", etagsMap(instances));
+}
+
+// detectEtagChanges compares the last set of etags stored in .firebaserc to the currently deployed etags,
+// and returns the ids on any instances have different etags.
+export function detectEtagChanges(
+  rc: RC,
+  projectId: string,
+  instances: { instanceId: string; etag?: string }[]
+): string[] {
+  const lastDeployedEtags = rc.getEtags(projectId).extensionInstances;
+  const currentEtags = etagsMap(instances);
+  // If we don't have a record of the last deployed state, detect no changes.
+  if (!lastDeployedEtags || !Object.keys(lastDeployedEtags).length) {
```

Comment:
If we don't have tags recorded but already have an exported manifest, and we deployed, that would override silently right? How does this compare to forcing them to export first before deploying?

Reply:
Unfortunately, if we don't have this if statement to skip when there is no recorded etags, we'll trigger this for _everyone_ on the first deployment on the version this releases in. I didn't want to inaccurately trigger this for everyone, so i figured it was better to only trigger if we actually have etags.

Note that we also have a separate check that warns if you have existing instances that don't match what you are deploying - this should cover some of these cases.